### PR TITLE
Run the package+deploy dialog if the stack settings is closed with Ok

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
If the dialog for stack settings is closed successfully, start the deployment
Also cleaned up some of the DialogWrapper code for the stack settings

## Testing
Manual

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt [hunk 1]
```diff
@@ -12,24 +12,40 @@ import com.intellij.openapi.vfs.VirtualFile
...
+        val deployDialog = SamDeployDialog(
+            project,
+            stackDialog.stackName,
+            templateFile,
+            stackDialog.region,
+            stackDialog.bucket
+        )
+
+        deployDialog.show()
+        deployDialog.executeDeployment()
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt]: Fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt [hunk 2]
```diff
@@ -12,24 +12,38 @@ import com.intellij.openapi.vfs.VirtualFile
...
+        val stackDialog = DeployServerlessApplicationDialog(project, template.parameters())
+        stackDialog.show()
+        if (!stackDialog.isOK) return
+
+        val deployDialog = SamDeployDialog(
+            project,
+            stackDialog.stackName,
+            templateFile,
+            stackDialog.region,
+            stackDialog.bucket
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt]: parameters?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt]: Updated PR with parameters

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt]:
```
    SAM CLI: /Users/alfredbr/Library/Python/3.7/bin/sam deploy --template-file packaged-template.yaml --stack-name SamDeployTest --capabilities CAPABILITY_IAM CAPABILITY_NAMED_IAM --no-execute-changeset --parameter-overrides TestParameter=FooBar
```

**Pull request merged**